migrations/lib/importer/discourse_db.rb

```diff
@@ -23,14 +29,28 @@ def copy_data(table_name, column_names, rows)
         @connection.transaction do
           @connection.copy_data(sql, @encoder) do
             sliced_rows.each do |row|
-              data = column_names.map { |c| row[c] }
+              if row[SKIP_ROW_MARKER]
+                skipped_rows << row
+                next
+              end
+
+              i = 0
+              while i < column_count
+                data[i] = row[column_names[i]]
+                i += 1
+              end
```

Comment:
The `while` loop should be faster than the original `map`. This might compensate for the additional work this block is doing now.